include/ddc/discrete_domain.hpp

```diff
@@ -63,6 +63,12 @@ class DiscreteDomain
     template <class...>
     friend class DiscreteDomain;
 
+    static_assert(
+            ((type_seq_size_v<type_seq_remove_t<detail::TypeSeq<DDims...>, detail::TypeSeq<DDims>>>
+              == sizeof...(DDims) - 1)
+             && ...),
+            "The dimensions of a DiscreteDomain must be unique");
```

Comment:
I think this is a good idea, could you make it a helper trait ? like `detail::type_seq_unique_v<detail::TypeSeq<DDims...>>` ?